testlapack/dlahr2.go

```diff
@@ -0,0 +1,83 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gonum/floats"
+)
+
+type Dlahr2er interface {
+	Dlahr2(n, k, nb int, a []float64, lda int, tau, t []float64, ldt int, y []float64, ldy int)
+}
+
+type Dlahr2test struct {
+	N, K, NB int
+	A        []float64
+
+	AWant   []float64
+	TWant   []float64
+	YWant   []float64
+	TauWant []float64
+}
+
+func Dlahr2Test(t *testing.T, impl Dlahr2er) {
```

Comment:
It's fine to test against the Lapack implementation, but the primary test should be of the function behavior itself. Using the Fortran code directly in Dlasq was because I did not understand the purpose of the routine, and so could not write an independent test for it. The behavior of this function seems clear, and so there should be a direct test of function behavior. There is no sense in deleting the test you already have, but perhaps give the zip file test a different name.